platform-sdk/docs/proposals/README.md

```diff
@@ -0,0 +1,150 @@
+# Platform Design Proposal Process
+
+This document describes the kind of code changes that require an accepted design proposal, the process for creating,
+reviewing, and merging platform design proposals, and how accepted proposals can be updated.
+
+The purpose of the design proposal process is to increase the quality of designs for larger and impactful changes
+before time is invested in the code. Higher quality designs reduce misdirected time investments and performance testing
+costs.
+
+The status of all design proposals can be viewed
+at [Platform Design Proposals](https://github.com/orgs/hashgraph/projects/73/views/1) project board.
+
+---
+
+## Design Proposal Criteria
+
+If code changes fall into one of the categories below, then an accepted design proposal is required before the code can
+merge.
+
+1. Publicly available inputs/outputs/API including disk
+2. Core behaviors
+3. Any non-trivial change that is expected or is likely to impact performance (some judgement must be applied)
+4. Changes to interactions between major systems or teams (platform, services, mirror node, devops, etc.)
+5. Module Structure & Repos
+
+---
+
+## Design Proposal Process Flow
+
+![](designProposalFlow.drawio.svg)
+
+## Creating A Proposal
+
+All proposals have an owner who is responsible for taking it through the process to completion. Proposals must be
+presented in the form of a PR containing the proposal in a subdirectory of `platform-sdk/docs/proposals/`. Proposals may
+contain diagrams and supporting materials. Diagrams must be created with DrawIO and exported in SVG format. All content
+must be editable (i.e. no images). The only exception is for images that are generated by code, like the wiring diagram.
```

Comment:
Why lock the tool used to generate diagrams? For example, there are easier and more automated tools than drawIO for generating UML diagrams. As providing a diagram is in service of the people reading the proposal, it should be up to the owner to choose which tool balances the effort and is more suitable for creating them case by case.

Reply:
Also, when using Intellij, it is easy to paste an image, and it automatically adds and links the file to the document in the position you pasted the image. It facilitates adding the image, but the images are generated in `PNG` format, not `SVG`

Reply:
The reason for naming drawIO as the diagram tool of choice is for two reasons:

1) consistency
2) it's a free tool available to all, and is therefore open-source friendly

The company made this decision a couple of years ago that all diagrams in the repo would be created with drawIO for these reasons.

Reply:
One of my secret desires is to create a new open alternative to draw.io that is both really great to use and has deep integration into github and IntelliJ. Sigh.

The core argument for Draw.io is that anyone can edit it. If we just take a PNG and check it in, then if someone else creates a PR to modify the diagram, they have to recreate it in their tooling to be able to edit it. I think the pain of using draw.io is much less than the pain of having to rebuild a diagram.

The wiring diagram (which is called out as an exception) is different because it can be auto-generated from source, so anyone can edit it.

Reply:
I'm already a victim of this policy as I used Lucid to draw a beautiful diagram of shining awesomeness, and now I will have to rebuild it in the dull colors and fonts of Draw.io. Boo.

But I agree with the policy despite that, because no-one else (outside SL) has a hope of contributing to that diagram. And that's wrong. So I should stop using Lucid. Sigh.

Reply:
Lucid can export diagrams in drawIO format =)

Reply:
I understand the reasoning. Still missing the possibility of using:
* https://plantuml.com/
* https://plugins.jetbrains.com/plugin/7017-plantuml-integration

Or even the diagrams generated by IntelliJ, seems like a loss.